lib/doorkeeper/oauth/refresh_token_request.rb

```diff
@@ -85,7 +85,10 @@ def validate_client_match
 
       def validate_scope
         if @original_scopes.present?
-          ScopeChecker.valid?(@original_scopes, refresh_token.scopes)
+          ScopeChecker.valid?(
+            scope_str: @original_scopes,
+            server_scopes: refresh_token.scopes
+          )
```

Comment:
shouldn't this also pass `grant_type: Doorkeeper::OAuth::REFRESH`?

Reply:
This is with the original implementation. I don't believe there is a scope that is only enabled on refresh.